src/main/java/org/opensearch/agent/tools/RAGTool.java

```diff
@@ -193,8 +187,30 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
             log.error("Failed to search index.", e);
             listener.onFailure(e);
         });
-        vectorDBTool.run(Map.of(VectorDBTool.INPUT_FIELD, embeddingInput), actionListener);
 
+        Map<String, Object> params = new HashMap<>();
+        params.put(VectorDBTool.INDEX_FIELD, this.index);
+        params.put(VectorDBTool.EMBEDDING_FIELD, this.embeddingField);
+        params.put(VectorDBTool.SOURCE_FIELD, gson.toJson(this.sourceFields));
+        params.put(VectorDBTool.MODEL_ID_FIELD, this.embeddingModelId);
+        params.put(VectorDBTool.DOC_SIZE_FIELD, String.valueOf(this.docSize));
+
+        switch (this.queryType) {
```

Comment:
How about moving the create NeuralSparseSearchTool/VectorDBTool logic to RAGTool.Factory.create? It is more logical consistent to create component during init process

Reply:
And if there is malformed parameters, it makes more sense to throw it from "create" instead of "run"

Reply:
good call. It works the same, but I agree that the users input error can be checked in the earlier stage. Move the subtools(NeuralSparseSearchTool/VectorDBTool) initiation in the factory  create session.

Reply:
Since we have moved subtools initiation in create session, we don't use the fields `index`, `embeddingField` ... in RAGTool.run now. We can remove these fields from RAGTool

Reply:
removed